### PR TITLE
Feature/roc 1794 pay by account

### PR DESCRIPTION
**Pull request by sabahirfan**
**JIRA:**
https://tools.hmcts.net/jira/browse/ROC-1794

**Fee Account payment Page**

### CODE REVIEW

File: src/main/app/forms/models/FeeAccount.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  @IsFeeAccount({ message: ValidationErrors.FEE_ACCOUNT_INVALID })
+  reference?: string
+
+  constructor (reference?: string) {
+    this.reference = reference
+  }
+
+  static fromObject (value?: any): FeeAccount {
+    if (value == null) {
+      return value
```
Comment (dharmendrak) [inline src/main/app/forms/models/FeeAccount.ts]: aren't we returning default contractor

File: src/main/features/claim/paths.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ export class Paths {
   static readonly defendantRepAddressPage = new RoutablePath('/claim/defendant-reps-address')
   static readonly statementOfTruthPage = new RoutablePath('/claim/statement-of-truth')
   static readonly payByAccountPage = new RoutablePath('/claim/pay-by-account')
+  static readonly claimSubmittedPage = new RoutablePath('/claim/claim-submitted')
```
Comment (dharmendrak) [inline src/main/features/claim/paths.ts]: change to /claim/submitted

File: src/main/features/claim/routes/pay-by-account.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    })
+}
+
+export default express.Router()
+  .get(Paths.payByAccountPage.uri, (req: express.Request, res: express.Response) => {
+    renderView(new Form(res.locals.user.legalClaimDraft.feeAccount.reference), res)
+  })
+  .post(Paths.payByAccountPage.uri, FormValidator.requestHandler(FeeAccount, FeeAccount.fromObject),
+    ErrorHandling.apply(async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
+      const form: Form<FeeAccount> = req.body
```
Comment (dharmendrak) [inline src/main/features/claim/routes/pay-by-account.ts]: add extra line

File: src/main/features/claim/views/pay-by-account.njk [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+{% from "form.njk" import csrfProtection, saveAndContinueButton, errorSummary, textInput %}
+
+{% block content %}
+
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <div class="task form-group">
+        <div class="column-half bold">{{ t('Total cost') }}</div>
+        <div class="column-half bold numeric">{{ feeAmount | numeral }}</div>
```
Comment (robertparkinson) [inline src/main/features/claim/views/pay-by-account.njk]: remove numeric, it doesn't do anything

Comment (sabahirfan) [inline src/main/features/claim/views/pay-by-account.njk]: Done.

File: src/main/features/claim/paths.ts (hunk 2 above)
Comment (sabahirfan) [inline src/main/features/claim/paths.ts]: Done

File: src/main/app/forms/models/FeeAccount.ts (hunk 1 above)
Comment (sabahirfan) [inline src/main/app/forms/models/FeeAccount.ts]: leaving it as discussed.

**Pull request merged**